src/lib.rs

```diff
@@ -1099,6 +1122,12 @@ pub enum Expression {
         left: Handle<Expression>,
         right: Handle<Expression>,
     },
+    /// Atomic function.
+    Atomic {
+        /// Pointer to an atomic value.
+        pointer: Handle<Expression>,
+        fun: AtomicFunction,
+    },
```

Comment:
We agreed in chat that these should not be expressions, since they're not pure. This is a major change, so I'm going to "request changes" in this PR.